cmd/srcd/cmd/sql.go

```diff
@@ -195,49 +169,72 @@ func pingDB(ctx context.Context, client api.EngineClient, queryTimeoutSeconds ti
 	}
 }
 
-func runQuery(client api.EngineClient, query string) error {
-	// Might have to pull some images
-	ctx, cancel := context.WithTimeout(context.Background(), 1440*time.Minute)
-	defer cancel()
+func installMysqlCli(client api.EngineClient) error {
```

Comment:
do you think `installMysqlCliWithDeps` is a good name for it?
I do everything in 1 function because this way I'm able to use `defer` and print log message only once.

Reply:
Makes sense to bundle it in one function. I guess what seemed a bit strange to me is here:
```go
if err := installMysqlCli(client); err != nil {
	fatal(err, "could not install mysql client")
}

connReady := logAfterTimeoutWithSpinner("waiting for gitbase to be ready", 5*time.Second, 0)
err = ensureConnReady(client)
connReady()
if err != nil {
	fatal(err, "could not connect to gitbase")
}
```

It reads like you install the mysql client, and then you check the connectivity to gitbase. Maybe this function should be called `startGitbaseWithClient`?

Honestly it's a small nickpick, leave as it is if you think it's ok.

Reply:
`startGitbaseWithClient` is a good name. Applied in:
https://github.com/src-d/engine/pull/353/commits/9542724d88be34a2d9aca693edf21274967a923e

also, I changed the signature so the function doesn't return an error anymore. I did it to make error handling for gitbase & mysql-cli consistent.

Reply:
Now I remember the original reason. I looked at how web client works. It removes container on exit. So I did the same for cli client.